pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -364,6 +379,33 @@ func hasChangedImmutableField(oldObject, newObject *hivev1.ClusterDeploymentSpec
 	return false, ""
 }
 
+// currently only allow controlPlaneConfig.servingCertificates to be mutable
+func hasChangedImmutableControlPlaneConfigFields(origObject, newObject *hivev1.ClusterDeploymentSpec) bool {
+	origCopy := origObject.ControlPlaneConfig.DeepCopy()
+	newCopy := newObject.ControlPlaneConfig.DeepCopy()
+
+	// blank out the servingCertificates, since we don't care if they're different
+	origCopy.ServingCertificates = hivev1.ControlPlaneServingCertificateSpec{}
+	newCopy.ServingCertificates = hivev1.ControlPlaneServingCertificateSpec{}
+
+	if !reflect.DeepEqual(origCopy, newCopy) {
+		return true
+	}
+
+	return false
+}
+
+func getOriginalMachinePool(origMachinePools []hivev1.MachinePool, name string) *hivev1.MachinePool {
```

Comment:
Is this used somewhere?

Reply:
yeah, in a completely different PR https://github.com/openshift/hive/pull/356 :) will remove it